tests/impls/descriptor_generator/test_prepopulated.py

```diff
@@ -0,0 +1,37 @@
+def test_prepopulated_descriptor_generator_plugin():
+    """
+    test that plugin and implementation is discoverable
+    """
+
+    try:
+        from smqtk_descriptors.impls.descriptor_generator.prepopulated import (
+            PrePopulatedDescriptorGenerator,
+        )
+
+        print("Module imported successfully")
+    except ImportError as e:
+        print(f"ImportError: {e}")
+
+    # print("Class info: ", PrePopulatedDescriptorGenerator)
+
+    import pkg_resources
```

Comment:
@Purg is there a section for test-time dependencies. I see `[tool.poetry.dev-dependencies]`, will that work on CI?

Although now that I'm looking at it, this module is part of the `setuptools` package, which we may not want to depend on? What is the correct way to `iter_entry_points` without setuptools?

Because functionality of these modules _require_ that they be installed and exposed via entry points, I think it is in scope to test that here.

Reply:
> @Purg is there a section for test-time dependencies. I see `[tool.poetry.dev-dependencies]`, will that work on CI?

That is the appropriate section for developer specific dependencies, which overlaps with a test-time usecase IMO. This section's dependencies will also be installed during CI. Anything `poetry install` is invoked, it is assumed a developer is running the command, so the dev deps are installed. There are of course options to not do this, but that is the default.

> 
> Although now that I'm looking at it, this module is part of the `setuptools` package, which we may not want to depend on? What is the correct way to `iter_entry_points` without setuptools?

`setuptools` is not required to peruse python entrypoints, which is one mechanism under the hood that finds plugins. This uses the now standard `importlib.metadata` package from 3.8 and beyond. Before 3.8, `importlib_metadata` the package is required by `smqtk-core`.

> 
> Because functionality of these modules _require_ that they be installed and exposed via entry points, I think it is in scope to test that here.

My take on the "unit scope" here is that diving into entrypoint existence is overreach. I would argue that the requirement is that the `PrePopulatedDescriptorGenerator` is *discoverable*, not specifically how it was discovered. The *how* is generally the responsibility of `smqtk-core` and its unit-testing. If you feel strongly that you want to enforce that entrypoints specifically are what provides this implementation, I would direct you towards the `smqtk_core.plugin.discover_via_entrypoint_extensions` function, which is what is used under `Pluggable.get_impls()`. There is a world in which the `Pluggable.get_impls` method is updated to accept parameters to indicate which methods of discovery are used or not, e.g. for this test, disable all but entrypoint-based discovery to ensure that means results in  your impl being returned (that's clearly a little bigger of a change).

Reply:
The unit test has been re-written with a different approach.